homework/index.js

```diff
@@ -42,67 +54,55 @@
     // manually create td with a link
     const td = createAndAppend('td', tr);
     createAndAppend('a', td, {
-      href: data[index].html_url,
-      text: data[index].name,
+      href: repository.html_url,
+      text: repository.name,
       target: '_blank',
     });
-    // a function to create td without link
-    function createTd(textKey, textContent) {
-      const parent = createAndAppend('tr', tbody);
-      createAndAppend('td', parent, {
-        text: `${textKey}: `,
-        class: 'label',
-      });
-      createAndAppend('td', parent, {
-        text: textContent,
-      });
-    }
-    createTd('Description', data[index].description);
-    createTd('Forks', data[index].forks);
-    createTd('Updated', data[index].updated_at);
+    createTd('Description', repository.description, tbody);
+    createTd('Forks', repository.forks, tbody);
+    createTd('Updated', repository.updated_at, tbody);
   }
 
   // contributors list
-  function listContributors(index, data, container) {
-    const contributorUrl = data[index].contributors_url;
+  function listContributors(repository, container) {
+    const contributorUrl = repository.contributors_url;
     fetchJSON(contributorUrl, (err, contributors) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
```

Comment:
I seem to miss the `return` statement. Without it, the code just continues with the next statement, which is wrong. Instead, your function needs to 'return' (that is what I mean with 'exit early'). Same thing in the other place in your code where you want to 'exit early'.

Reply:
I have solved this too.
Thanks a lot